onnxscript/rewriter/function_rule.py

```diff
@@ -185,57 +175,53 @@ def try_rewrite_function(
             )
 
         if not self._match_function(function, pkg_name):
-            return False
+            return None
         logger.info(
             "Rule %s matched function %s::%s",
             self.__class__.__name__,
             function.domain,
             function.name,
         )
-
         try:
-            new_function, opset_imports = self.compose_new_function(function, pkg_version)
+            new_function = self.compose_new_function(function, pkg_version)
         except FunctionRewriteError as e:
             logger.warning("Could not rewrite function: %s", e)
-            return False
-
-        nodes = new_function.node
+            return None
 
-        del function.input[:]
-        function.input.extend(new_function.input)
-        del function.output[:]
-        function.output.extend(new_function.output)
+        new_function.name = function.name
+        new_function.domain = function.domain
 
-        del function.node[:]
-        function.node.extend(nodes)
-        for new_opset in opset_imports:
-            function.opset_import.append(new_opset)
-            if new_opset.domain not in self._opset_imports:
-                model.opset_import.append(new_opset)
```

Comment:
This PR has a regression here. I didn't add the new opset version into opset_imports..

Reply:
@justinchuby Even with replace the whole new function, I think we still have to manually update opset_imports to model.opset_imports, right?

Reply:
That's right. We could automate this process though.

Reply:
https://github.com/microsoft/onnxscript/issues/1444